sessionagent/session_agent_test.go

```diff
@@ -0,0 +1,82 @@
+// -*- Mode: Go; indent-tabs-mode: t -*-
+
+/*
+ * Copyright (C) 2019 Canonical Ltd
```

Comment:
this file seems to be misnamed, it's really rest_api_test.go ?

Reply:
This was intended as a test for the setup/shutdown of the HTTP server.  I needed to make some request in the test, and we've only got a single trivial method.

I expect we'll add new tests in a rest_api_test.go later, but they would likely be at the level of the `agent.Command` API rather than through the full HTTP stack.

Reply:
I've added a direct test for `sessionInfo`, and made it more explicit that the tests in `session_agent_test.go` are about testing the HTTP implementation itself: i.e. one test for Start/Stop, and another that Dying() works.